python/cudf/cudf/tests/conftest.py

```diff
@@ -15,6 +19,64 @@ def datadir():
     return pathlib.Path(__file__).parent / "data"
 
 
+@pytest.fixture(
+    params=[None, 42, np.random.RandomState, cp.random.RandomState]
+)
+def random_state_tuple(request):
+    """
+    Specific to `test_dataframe_sample*` and `test_series_sample*` tests.
+    A pytest fixture of valid `random_state` parameter pairs for pandas
+    and cudf. Valid parameter combinations, and what to check for each pair
+    are listed below:
+
+    pandas:   None,   seed(int),  np.random.RandomState,  np.random.RandomState
+    cudf:     None,   seed(int),  np.random.RandomState,  cp.random.RandomState
+    ------
+    check:    shape,  shape,      exact result,           shape
+
+    Each column above stands for one valid parameter combination and check.
+    """
+
+    def shape_checker(expected, got):
+        assert expected.shape == got.shape
+
+    def exact_checker(expected, got):
+        assert_eq(expected, got)
+
+    seed_or_state_ctor = request.param
+    if seed_or_state_ctor is None:
+        return None, None, shape_checker
+    elif isinstance(seed_or_state_ctor, int):
+        return seed_or_state_ctor, seed_or_state_ctor, shape_checker
+    elif seed_or_state_ctor == np.random.RandomState:
+        return seed_or_state_ctor(42), seed_or_state_ctor(42), exact_checker
```

Comment:
If we're just going to hardcode the values, can we put that directly in the parametrization? Or will that lead to the same object getting used every time and cause problems for tests?

Reply:
> Or will that lead to the same object getting used every time and cause problems for tests?

Yes I think that's what happens. The same random state is returned for both cudf and pandas. After used in pandas the iteration is incremented and this leads to different rows to sample from in cuDF. That's why we are to construct a new random state every time the fixture is requested.

Reply:
Could we maybe use a lambda somehow? It would be nice if all the parametrization information would be capture in the fixture params, rather than having some of them there and some of the values inlined like the 42 here.

Reply:
Somehow (possibly due to the previous refactor) I realized this function is no longer needed, I'm in-lining everything in the fixture parameter list.